src/sha256_ni_rec.cpp

```diff
@@ -0,0 +1,286 @@
+/*
+ * sha256_ni_rec.cpp
+ *
+ *  Created on: May 28, 2023
+ *      Author: mad
+ */
+
+// purpose: rewrite of generic sha256_ni into optimized recursive_sha256_ni
+// prerequisite: length is 32 bytes (recursive sha256)
+// optimization: rewrite with prerequisite to help compiler optimizations
+
+#include <sha256_ni_rec.h>
+
+#include <cstring>
+#include <stdexcept>
+
+#if defined(__SHA__) || defined(_WIN32)
+
+#include <immintrin.h>
+
+#ifdef _WIN32
+#include <intrin.h>
+#endif
+
+#ifdef _WIN32
+#define optimalinline inline
```

Comment:
Macros should be upper case, like `OPTIMALINLINE`

Reply:
Fixed & commited.